### PR TITLE
add Badge Component

### LINKED ISSUE #4: <Badge />
### Figma design
https://www.figma.com/file/0vYcLbHGHFIJ44hFI45WQJ/%F0%9F%9A%80-Habitat?node-id=9248%3A11048&mode=dev

### Props
| Name | Type | Default | is optional? | other values |
|--|--|--|--| --|
| status | `string` | undefined | yes | "warning", "error", "success |
| icon | `component` | | yes | |
| theme | `string` | "light" | yes | "dark" |

### Notes
the badge text must be as children not as prop

**the "color" part in the Figma file can be ignored**

### LINKED ISSUE #13: storybook controls
for some reason before being able to interact with a control in Storybook a "set [type]" button appears on top of the control itself, as shown in the image below

image

is this some misconfiguration or bug introduced by me? 
please help me understand the cause and fix

### PR DESCRIPTION
**Pull request by Serse2**
## Figma reference

Link [here](https://www.figma.com/file/0vYcLbHGHFIJ44hFI45WQJ/%F0%9F%9A%80-Habitat?node-id=11686%3A26095&mode=dev)

## What

Start create basic badge and status badge, for the moment the notification badge is out of this component, need discuss??

##  Notes

The icon for now are a React.Component, i've try to do with a plugin for vite but it gives me a lot of error, i need to investigate why the `vite-plugin-svgr` not working properly.

This will close #4
This will close #13 

## Release checklist

- [x] assign this PR to yourself
- [x] add meaningful labels
- [x] add at least 2 reviewers

(_Chromatic approvals, unit tests outcome and devs approval are already verified automatically_)

### CODE REVIEW

File: src/stories/Components.Badge.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// Button.stories.ts|tsx
```
Comment (pietrograndi) [inline src/stories/Components.Badge.stories.tsx:1]:
```suggestion
// Badge.stories.ts|tsx
```

File: src/components/Badge/Badge.css.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/* eslint-disable sort-keys-fix/sort-keys-fix */
+import { recipe } from "@vanilla-extract/recipes";
+import { vars } from "../../utils/theme.css";
+
+export const BadgeContainer = recipe({
+  base: {
+    alignItems: "center",
+    borderRadius: 20,
+    display: "inline-flex",
+    fontFamily: "Inter, -apple-system, BlinkMacSystemFont, sans-serif",
```
Comment (GiovanniVisentiniCasavo) [inline src/components/Badge/Badge.css.ts:10]: can we use some habitat typography to configure the font here?

File: src/stories/Components.Badge.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+import type { Meta, StoryObj } from "@storybook/react";
+import { Badge } from "../components/Badge/Badge";
+import { HabitatTheme } from "../utils/theme.css";
+import { BadgeIcon } from "../assets/badge/BadgeIcon";
```
Comment (GiovanniVisentiniCasavo) [inline src/stories/Components.Badge.stories.tsx:6]: the badge icon is used only in this test, can we inline it in this file?

File: src/components/Badge/Badge.css.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    fontWeight: 400,
+    gap: 8,
+    justifyContent: "center",
+    lineHeight: 1.5,
+    padding: "8px 16px",
+  },
+  variants: {
+    theme: {
+      dark: {
+        backgroundColor: vars.colors.deepgreen[100],
```
Comment (GiovanniVisentiniCasavo) [inline src/components/Badge/Badge.css.ts:21]: it is ok to use directly the color name here, and not having a token?

File: src/components/Badge/Badge.css.ts (hunk 2 above)
Comment (moebiusmania) [reply src/components/Badge/Badge.css.ts:10]: @GiovanniVisentiniCasavo yep, I'm still working on it that why for the moment is hard-coded

File: src/components/Badge/Badge.css.ts (hunk 4 above)
Comment (moebiusmania) [reply src/components/Badge/Badge.css.ts:21]: we were checking it with @Serse2 , on Figma most of the stuff uses the primitive tokens rather than the domain ones, as overall "agreement" of the library we should take the info from the Figma as "truth" and eventually push the designers to update it if something isn't right.

TL;DR we shouldn't "guess" which token is the correct one

File: src/stories/Components.Badge.stories.tsx (hunk 1 above)
Comment (Serse2) [reply src/stories/Components.Badge.stories.tsx:1]: Yes you are right, i'll fix this in the next commit

File: src/stories/Components.Badge.stories.tsx (hunk 3 above)
Comment (Serse2) [reply src/stories/Components.Badge.stories.tsx:6]: Good point, i'll move the BadgeIcon inside the stories component because is use only to show the possibility to insert an icon inside the component.

**Review by moebiusmania — APPROVED** [review]

**Comment by moebiusmania** [comment]
PR is fine, using the <Body size="s" /> within the component introduces some undesired margins, but this is due to the lack of reset in Storybook or the library itself. We will fix this in a follow up PR

File: src/components/Badge/Badge.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  icon?: React.ReactNode;
+  status?: "success" | "error" | "warning";
+  theme?: "light" | "dark";
+};
+
+type StatusIcon = {
+  error: React.ReactNode;
+  success: React.ReactNode;
+  warning: React.ReactNode;
+};
```
Comment (pietrograndi) [inline src/components/Badge/Badge.tsx:19]:
```suggestion
type StatusIcon = {
 [key in BadgeProps['status']]: React.ReactNode
};
```
@Serse2 what do you think?

**Review by pietrograndi — APPROVED** [review]
LGTM, feel free to not consider my 2 cents on `Badget.tsx`

File: src/components/Badge/Badge.tsx (hunk 5 above)
Comment (Serse2) [reply src/components/Badge/Badge.tsx]: Good point! i change che type like this

`type Status = "success" | "error" | "warning";`

```
type StatusIcon = {
  [key in Status]: React.ReactNode;
};
```

**Review by moebiusmania — APPROVED** [review]

**Review by pietrograndi — APPROVED** [review]
